Eastern-Research-Group/mywaterway
# Feature/670 display watershed size

## Related Issues:
* [HMW-670](https://jira.epa.gov/browse/HMW-670)

## Main Changes:
* Added popups for the Watersheds layer and the Boundaries layer.
  * These popups are identical: if both are visible and clicked, only one will be shown.
* Removed the default handling of popups universally.
  * Previously, this was only done on the Tribal page and the Monitoring tab, but the Watersheds layer is available everywhere, and we'd like the watershed popups to always be last. To my knowledge, this only affected the Protected Areas layer (a MapImageLayer) popups, which need to fetch from the server before showing. I changed this to instead fetch the features manually if the layer is visible. This also fixes the popups in the two locations mentioned previously. I wasn't able to recreate the Esri cursor spinner, though, so the default browser wait cursor is used instead.
* Added the watershed area (in acres and square kilometers) to the watershed summary information to the right of the pin icon above the tabs on the Community page.
* Updated the Upstream Watershed popup to use two points of precision for consistency with the other watersheds popup.

## Steps To Test:
1. Go to http://localhost:3000/community/dc/overview.
2. Click outside the Boundaries layer. The "Change to This Location?" popup should appear.
3. Toggle on the visibility of the Watersheds layer, then click outside the Boundaries layer again. A watershed popup should show with a "Change to this location?" button within.
4. With the Watersheds layer still visible, click inside the boundaries of the current HUC. Only one watershed popup should show, and no "Change to this location?" message should be displayed.
5. Toggle on the visibility of the Protected Areas layer, then click on of its features. The cursor should change to a wait cursor inside the map container, then a popup should show with one or more Protected Areas features.
6. Repeat steps 1-5 for other pages.
